raitools/raitools/_managers/counterfactual_manager.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the Counterfactual Manager class."""
+
+from raitools._managers.base_manager import BaseManager
+
+
+class CounterfactualManager(BaseManager):
+    def __init__(self):
+        pass
+
+    def add(self,
+            continuous_features=None,
+            outcome_name=None,
+            total_CFs=None,
+            desired_range=None,
+            desired_class="opposite",
+            permitted_range=None,
+            features_to_vary=None,
+            method=None):
+        raise NotImplementedError("Add is not implemented")
+
```

Comment:
Can we special case the message per manager? Like "Add for counterfactual manager is not implemented"? It would be difficult to figure out from the exception message as to which manager has not implemented.

Reply:
done!